nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/utils/ComponentStateCheckpointStore.java

```diff
@@ -0,0 +1,228 @@
+                checkpoint.getConsumerGroup(),
+                checkpoint.getPartitionId()
+        );
+        if (offset == null) {
+            newState.remove(key);
+        } else {
+            newState.put(key, offset.toString());
+        }
+
+        try {
+            state.replaceState(oldState, newState);
+        } catch (IOException e) {
+            return Mono.error(e);
+        }
+
+        // Note that we do not commit here because there is an implicit
+        // agreement that the process session factory will provide a session
+        // that is automatically committed (this is done by the processor).
+        return Mono.empty();
+    }
+
+    private Checkpoint convertCheckpoint(PartitionContext context, String value) {
+        return new Checkpoint()
+                .setFullyQualifiedNamespace(context.getFullyQualifiedNamespace())
+                .setEventHubName(context.getEventHubName())
+                .setConsumerGroup(context.getConsumerGroup())
+                .setPartitionId(context.getPartitionId())
+                .setOffset(Long.parseLong(value));
+    }
+
+    private PartitionOwnership convertOwnership(PartitionContext context, String value) {
+        final String[] parts = value.split("/", 2);
+        if (parts.length != 2) {
+            throw new ProcessException(String.format("Invalid ownership value: %s", value));
+        }
+        return new PartitionOwnership()
+                .setFullyQualifiedNamespace(context.getFullyQualifiedNamespace())
+                .setEventHubName(context.getEventHubName())
+                .setConsumerGroup(context.getConsumerGroup())
+                .setPartitionId(context.getPartitionId())
+                .setETag(value)
+                .setOwnerId(parts[0])
+                .setLastModifiedTime(Long.parseLong(parts[1])
+                );
+    }
+
+    private <T> List<T> getEntries(final Map<String, String> map, String kind, BiFunction<PartitionContext, String, T> extract) throws ProcessException {
+        final List<T> result = new ArrayList<>();
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            final String key = entry.getKey();
+            final String[] parts = key.split("/", 5);
+            if (parts.length != 5) {
+                throw new ProcessException(
+                        String.format("Invalid %s key: %s", kind, entry.getKey())
+                );
+            }
+            if (!parts[0].equals(kind)) {
+                continue;
+            }
+            final String fullyQualifiedNamespace = parts[1];
+            final String eventHubName = parts[2];
+            final String consumerGroup = parts[3];
+            final String partitionId = parts[4];
+            PartitionContext partitionContext = new PartitionContext(
+                    fullyQualifiedNamespace,
+                    eventHubName,
+                    consumerGroup,
+                    partitionId
+            );
```

Comment:
Recommend pulling this logic out to a separate method so the formatting approach is clear.